router-tests/headers_test.go

```diff
@@ -227,3 +228,230 @@ func TestForwardHeaders(t *testing.T) {
 		})
 	})
 }
+
+func TestForwardRenamedHeaders(t *testing.T) {
+	t.Parallel()
+
+	const (
+		// Make sure you copy these to the struct tag in the subscription test
+		headerNameInGlobalRule     = "foo"
+		headerNameInSubgraphRule   = "barista" // This matches the regex in test1 subgraph forwarding rules
+		headerRenameInGlobalRule   = "light"
+		headerRenameInSubgraphRule = "cell" // This matches the regex in test1 subgraph forwarding rules
+		headerValue                = "bar"
+		headerValue2               = "baz"
+
+		subscriptionForGlobalRulePayload   = `{"query": "subscription { headerValue(name:\"light\", repeat:3) { value }}"}`
+		subscriptionForSubgraphRulePayload = `{"query": "subscription { headerValue(name:\"cell\", repeat:3) { value }}"}`
+	)
+
+	headerRules := config.HeaderRules{
+		All: config.GlobalHeaderRule{
+			Request: []config.RequestHeaderRule{
+				{
+					Operation: config.HeaderRuleOperationPropagate,
+					Named:     headerNameInGlobalRule,
+					Rename:    headerRenameInGlobalRule,
+				},
+			},
+		},
+		Subgraphs: map[string]config.GlobalHeaderRule{
+			"test1": {
+				Request: []config.RequestHeaderRule{
+					{
+						Operation: config.HeaderRuleOperationPropagate,
+						Matching:  "(?i)^bar.*",
+						Rename:    headerRenameInSubgraphRule,
+					},
+				},
+			},
+		},
+	}
+
+	t.Run("HTTP", func(t *testing.T) {
```

Comment:
Please also add a test that validates if "default" works with rename. Thank you.